src/storage.rs

```diff
@@ -161,9 +176,24 @@ where
     }
 
     // @TODO specify more useful error type
-    #[inline]
     fn init_active_blob(&mut self) -> Result<(), ()> {
-        self.active_blob = Some(Box::new(Default::default()));
+        let new_blob_name = format!(
+            "{}.{}.{}",
+            self.config.blob_file_name_prefix.as_ref().ok_or(())?,
+            self.blobs.len(),
+            BLOB_FILE_ETENSION
+        );
+        let path = self
+            .config
+            .work_dir
+            .as_ref()
+            .ok_or(())?
+            .join(&new_blob_name);
+        let mut runtime = tokio::runtime::Runtime::new().unwrap();
+        // @TODO remove block on
+        let blob = runtime.block_on(Blob::open_new(path)).unwrap();
```

Comment:
Why did you created an asynchronous `open` method and now just block on it? You should make the whole code asynchronous or make the `open` method synchronous.

Reply:
Outdated